README.md

```diff
@@ -104,7 +104,98 @@ module.exports = {
 
 Note that while the above example tests a webpage with [puppeteer](https://github.com/GoogleChrome/puppeteer) and uses pentf's has native support for HTTP requests (in `net_utils`) and email sending (in `email`), tests can be anything – they just have to fail the promise if the test fails.
 
-Have a look in the [API documentation](https://boxine.github.io/pentf/index.html) for various helper functions.
+Have a look in the [API documentation](https://boxine.github.io/pentf/) for various helper functions.
+
+## Tips for writing good tests
+
+By their very nature, end-to-end tests can be flaky, i.e. sometimes succeed and sometimes fail when run multiple times. We want the tests to only relay the flakiness of the systems we test, and not introduce any additional flakiness ourselves. Here are a few tips for that:
+
+### Use `data-testid` attributes in browser tests
+
+Unless otherwise documented, class names and document structures are subject to change. By setting an explicit `data-testid="foo"` attribute in the code, the developer and tester set up a stable contract.
```

Comment:
Nit: What do you think about changing the example to `data-testid="comment-button"` here, so that it's consistent with the example in the "use" section?

Reply:
Ok, updated.